src/shared/nested-dropdown-list/list-item.tsx

```diff
@@ -0,0 +1,209 @@
+        );
+    };
+
+    const hasExceededContainer = (item: FItemOption<V1, V2, V3>) => {
+        const displayText = listExtractor
+            ? listExtractor(item)
+            : item.toString();
+
+        let widthOfElement = 0;
+        if (labelRef && labelRef.current) {
+            widthOfElement = labelRef.current.getBoundingClientRect().width;
+        }
+
+        return StringHelper.shouldTruncateToTwoLines(
+            displayText,
+            widthOfElement
+        );
+    };
+
+    const renderListItem = () => {
+        if (!expand) return;
+
+        const nextSubItems = item.subItems.values();
+
+        return (
+            <ul data-testid="sub-category-list">
+                <List>
+                    <ul>
+                        {[...nextSubItems].map((item) => (
+                            <ListItem
+                                key={item.key}
+                                parentKeys={[...parentKeys, item.key]}
+                                item={item}
+                                mode={mode}
+                                listExtractor={listExtractor}
+                                selectedItems={selectedItems}
+                                selectedKeys={selectedKeys}
+                                defaultExpandKeys={defaultExpandKeys}
+                                visible={visible}
+                                onBlur={onBlur}
+                                onExpand={onExpand}
+                                onSelect={onSelect}
+                            />
+                        ))}
+                    </ul>
+                </List>
+            </ul>
+        );
+    };
+
+    if (!item.subItems) {
+        const hasExceeded = hasExceededContainer(item);
+
+        /**
+         * force re-render dropdown height
+         * buggy happened if flat option in tier 1
+         */
+        if (!hasExceeded) onExpand();
+
+        return (
+            <li ref={labelRef}>
+                <ListItemSelector
+                    type="button"
+                    data-testid="list-item"
+                    tabIndex={visible ? 0 : -1}
+                    $selected={checkListItemSelected(item.value)}
+                    onBlur={handleBlur}
+                    onClick={handleSelect}
+                >
+                    <Label $truncateType={itemTruncationType}>
+                        {itemTruncationType === "middle" && hasExceeded
+                            ? renderTruncatedText(item)
+                            : item.label}
+                    </Label>
+                </ListItemSelector>
+            </li>
+        );
+    }
+
+    return (
+        <li data-testid="category-list">
+            <ul>
+                <Category data-testid="category-title" onClick={handleClick}>
+                    <ArrowButton $expand={expand}>
+                        <TriangleIcon />
+                    </ArrowButton>
+                    <Title>{item.label}</Title>
```

Comment:
hmm category title may need truncation as well

Reply:
remove `<Title>` styled?

Reply:
ok. will update it once get back from designer.

Reply:
they only requested for `end` truncation for now. 

i updated to Liting, we provided `middle` and `end` to consumer and default is `end`